src/core/cgroup.c

```diff
@@ -1808,6 +1835,30 @@ static bool unit_has_mask_realized(
                 u->cgroup_invalidated_mask == 0;
 }
 
+static bool unit_has_mask_disables_realized(
+                Unit *u,
+                CGroupMask target_mask,
+                CGroupMask enable_mask) {
+
+        assert(u);
+
+        return u->cgroup_realized &&
+                FLAGS_SET(u->cgroup_realized_mask, target_mask) &&
+                FLAGS_SET(u->cgroup_enabled_mask, enable_mask);
+}
+
+static bool unit_has_mask_enables_realized(
+                Unit *u,
+                CGroupMask target_mask,
+                CGroupMask enable_mask) {
+
+        assert(u);
+
+        return u->cgroup_realized &&
+                (u->cgroup_realized_mask | target_mask) == u->cgroup_realized_mask &&
+                (u->cgroup_enabled_mask | enable_mask) == u->cgroup_enabled_mask;
+}
```

Comment:
i have the suspicion that both these calls should mask things for CGROUP_MASK_V1 in the case of the realization mask and  CGROUP_MASK_V2 for the enable mask, no? I mean, both masks might contain bpf pseudo-controllers bits set, and we really shouldn't get confused by that here...

Reply:
See how unit_has_mask_realized() does that

Reply:
i would love an explanatory comment added in to both functions, in particular how they compare with unit_has_mask_realized()